apps/member/views.py

```diff
@@ -48,35 +48,45 @@ class RecordsView(LoginRequiredMixin, DetailView):
     model = Member
     template_name = "records.html"
     default_resource_name = 'sharemyhealth'
-    default_record_type = 'all'
+    default_record_type = 'Condition'
 
     def get_context_data(self, **kwargs):
         """Add records data into the context."""
         # Get the data for the member, and set it in the context
-        data = get_member_data(
+        results = get_member_data(
             self.request.user,
             kwargs.get('object'),
             self.default_resource_name,
             self.default_record_type
         )
-        kwargs.setdefault('data', data)
-
-        # TODO: remove this line, but keep it here for now until get_member_data()
-        # returns meaningful data, so the template doesn't look blank.
-        kwargs.setdefault('records_options', RECORDS)
-
+        kwargs.setdefault('results', results)
         return super().get_context_data(**kwargs)
 
 
 class DataSourcesView(LoginRequiredMixin, DetailView):
     model = Member
     template_name = "data_sources.html"
-    default_record_type = 'all'
 
-    def dispatch(self, request, *args, **kwargs):
-        self.resource_name = kwargs.get('resource_name')
-        self.record_type = kwargs.get('record_type') or self.default_record_type
-        return super().dispatch(request, *args, **kwargs)
+    def get(self, *args, **kwargs):
+        """
+        The request.user may see the member's data sources if:
+         - the request.user is the member, or
+         - the request.user is in an Organization that has been granted access
+           to the member's data
+        """
+        member = get_object_or_404(Member.objects.all(), pk=kwargs['pk'])
```

Comment:
since this essentially a permissions check it would be nice to encapsulate it. Something like a [`user_passes_test`](https://docs.djangoproject.com/en/2.2/topics/auth/default/#limiting-access-to-logged-in-users-that-pass-a-test) check possibly.

Reply:
Sounds like a good idea